core/dbt/version.py

```diff
@@ -130,7 +130,11 @@ def _get_plugin_msg_info(
 
     needs_update = False
 
-    if plugin.major != core.major or plugin.minor != core.minor:
+    assert plugin.major is not None
+    assert plugin.minor is not None
+    assert core.major is not None
+    assert core.minor is not None
+    if plugin.major != core.major or plugin.minor < core.minor:
```

Comment:
So I talked with @colin-rogers-dbt + @mikealfare about this. I think we can get safely rid of this check entirely. Basically, we've decoupled the plugin (adapter specific) version and the core version. The majors no longer need to be equal, and the minor can be behind the core minor. The later might not actually be that uncommon.

All of this is the case because PyPI now handles this via `dbt-adapters`. Essentially, `dbt-core` specifies what `dbt-adapters` versions it is compatible with, and each dbt adapter also specifies what `dbt-adapters` versions it is compatible with. The result being, you'd have to intentionally force install incompatible versions for this check to be relevant. Thus, this check is unneeded.

Reply:
even better! you want to do the honors or should i?